packages/react/src/components/Slider/Slider-test.js

```diff
@@ -1,441 +1,489 @@
-    });
+      it('gracefully tolerates empty event passed to onChange', () => {
+        const { type, tab } = userEvent;
+        renderSlider({
+          ariaLabelInput: inputAriaValue,
+          value: initialValue,
+          max: 100,
+          onChange,
+        });
+        const inputElement = screen.getByLabelText(inputAriaValue);
+        tab(); // Brings focus to slider
+        tab(); // Brings focus to input
+        type(inputElement, '{space}');
+        tab(); // Brings focus out of input
+        expect(onChange).not.toHaveBeenCalled();
+      });
 
-    it('sets correct state when typing a valid value in input field', () => {
-      const evt = {
-        target: {
-          value: '12',
-        },
-      };
-
-      wrapper.instance().onChange(evt);
-      expect(wrapper.state().value).toEqual(12);
-      expect(handleChange).toHaveBeenLastCalledWith({ value: 12 });
-    });
-  });
+      it('should call onBlur as expected', () => {
+        const { type, tab } = userEvent;
+        renderSlider({
+          ariaLabelInput: inputAriaValue,
+          value: 10,
+          max: 100,
+          onBlur,
+        });
+        const inputElement = screen.getByLabelText(inputAriaValue);
+        tab(); // Brings focus to slider
+        tab(); // Brings focus to input
+        type(inputElement, '{space}');
+        tab(); // Brings focus out of input
+        expect(onBlur).toHaveBeenCalledTimes(2);
+      });
 
-  describe('syncs invalid state and props', () => {
-    const handleChange = jest.fn();
-    const handleRelease = jest.fn();
-    const handleBlur = jest.fn();
-
-    const wrapper = mount(
-      <Slider
-        id="slider"
-        className="extra-class"
-        value={50}
-        min={0}
-        max={100}
-        step={1}
-        onChange={handleChange}
-        onRelease={handleRelease}
-        onBlur={handleBlur}
-        invalid={false}
-      />
-    );
-
-    it('overrides invalid state for invalid prop', () => {
-      const changeEvt = {
-        target: {
-          value: '200',
-        },
-      };
-
-      const blurEvt = {
-        target: {
-          checkValidity: () => false,
-        },
-      };
-
-      wrapper.instance().onChange(changeEvt);
-      wrapper.instance().onBlur(blurEvt);
-      expect(wrapper.state().value).toEqual(200);
-      expect(wrapper.state().isValid).toEqual(true);
-      expect(
-        wrapper
-          .find(`.${prefix}--slider-text-input`)
-          .hasClass(`${prefix}--text-input--invalid`)
-      ).toEqual(false);
-    });
-  });
+      it('should call onKeyDown as expected', () => {
+        const { type, click } = userEvent;
+        renderSlider({
+          ariaLabelInput: inputAriaValue,
+          onChange,
+        });
+        const theSlider = screen.getByRole('slider');
+        click(theSlider);
+        type(theSlider, '{arrowright}');
+        type(theSlider, '{arrowright}');
+        expect(onChange).toHaveBeenCalledTimes(2);
```

Comment:
Seems like the initial click for focus is causing an `onChange`. Can we test `onKeyDown` instead? Or do we just change the expected to `3`

Reply:
Ah, thanks for the suggestion! `onKeyDown` seems to be working